### DISCUSSION TITLE
feature file size limitation

### DISCUSSION THREAD

**Discussion by PaskoTo** [reqnroll/Reqnroll#187] [Q&A]
I was unable to find this question, while I was searching on how to build out my tests. I am using VS2022 and I have found that when the feature file has about 1200 scenarios, the auto-generation of the .cs file causes a stack overflow error.

Is this a known limitation on feature file size? Is this based on scenario count, or file size?

I am autogenerating feature files and I can create them within any constraints, just need to know what they are.

Was old specflow user [years ago], new to reqnroll.

**Comment by gasparnagy** [comment]
Hello @PaskoTo! Welcome here 🤘!

There is no explicit scenario limit for a feature file (of course the bigger the slower). (At least not any that I am aware of.) The stack overflow error definitely sounds strange, because more scenario should not cause more recursion (that is usually the root cause of stack overflow).

I would suggest making a simple console app and try to load such big files with the [Gherkin parser](https://www.nuget.org/packages/Gherkin). That will give you an impression about the limit and maybe you can also get a stack trace of the exceptions you get.

As far as I remember, I have tested it with 100 scenarios per file and that was still OK.

**Reply by clrudolphi** [reply]
I've used the test sample generator in Reqnroll.VisualStudio to generate a feature file with 1300 scenarios and it generated the scenario code successfully.
It would be very helpful if @PaskoTo can provide a sample that reproduces the problem.

**Comment by PaskoTo** [comment]
I did some more research and it sounds like it is a memory management issue. If I have 3 feature files of scenarios [3k+] they all work fine. If I add all into 1 feature file - I see events:
* vbcscompiler.exe clr.dll application error
*  devenv.exe clr.dll application error
*  csc.exe clr.dll application error

When I open VS & try and open the sln - VS closes and adds events to event viewer.

Has anyone else seen this?

thnx,

tom

**Comment by PaskoTo** [comment]
as far as sample - what would you need. feature file is pretty basic, would you need step definitions as well?

**Reply by clrudolphi** [reply]
Would not need the step definition code if the problem can be recreated without them during compilation.
